### PR TITLE
custom vision scorer

### LINKED ISSUE #1635: [FEAT][Vision] allow use of sklearn metrics for classification
**Is your feature request related to a problem? Please describe.**
Ignite have a relatively small number of metric in oppose to sklearn. In addition a uniform API between tabular and classification metrics API is a big plus.

**Describe the solution you'd like**
Update vision class performance for vision classification to support the tabular metrics for classification.
i.e after digesting all batches and collection true labels, probabilities and predictions invoke the relevant sklearn metrics API in the same manner it is done in the tabular performance report.

### PR DESCRIPTION
**Pull request by JKL98ISR**
closes #1635

### CODE REVIEW

File: deepchecks/vision/metrics_utils/classification_scorer.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    """Abstract class to calculate average precision and recall for various vision tasks.
+
+    Parameters
+    ----------
+    score_func: Callable, default: None
+        Score function (or loss function) with signature `score_func(y_true, y_pred, **kwargs)`
+    needs_proba: bool, default: False
+        Whether score_func requires the probabilites or not.
+    **kwargs
+        Additional parameters to be passed to score_func.
```
Comment (ItayGabbay) [inline deepchecks/vision/metrics_utils/classification_scorer.py:30]: Add example to how to use it

**Review by ItayGabbay — APPROVED** [review]
